### ISSUE TITLE
On Adding new Job: 1235, "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'"

### ISSUE THREAD

**Issue by luxoar** [chrisspen/django-chroniker#90]
Hi,

I just tried django-chroniker for the first time and I like the possibility to manage cronjobs via the admin interface. Sadly I am getting the following error whenever I want to add a job via admin.
Can't seem to pin the problem here. Is django-chroniker compatible with MySQL?

Traceback

```
Environment:


Request Method: POST
Request URL: http://127.0.0.1:8000/admin/chroniker/job/add/

Django Version: 1.10.2
Python Version: 3.5.2
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'django.contrib.sitemaps',
 'django_comments',
 'django_comments_xtd',
 'rosetta',
 'compressor',
 'Bildungsprojekt',
 'easy_pdf',
 'widget_tweaks',
 'haystack',
 'robots',
 'CustomerService',
 'chroniker']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'projectaxel.middleware.MyMiddleware']



Traceback:

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/mysql/base.py" in execute
  110.             return self.cursor.execute(query, args)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in execute
  250.             self.errorhandler(self, exc, value)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/connections.py" in defaulterrorhandler
  42.         raise errorvalue

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in execute
  247.             res = self._query(query)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in _query
  411.         rowcount = self._do_query(q)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in _do_query
  374.         db.query(q)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/connections.py" in query
  270.             _mysql.connection.query(self, query)

The above exception ((1235, "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")) was the direct cause of the following exception:

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/contrib/admin/options.py" in wrapper
  544.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/contrib/admin/sites.py" in inner
  211.             return view(request, *args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/contrib/admin/options.py" in add_view
  1509.         return self.changeform_view(request, None, form_url, extra_context)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/usr/lib/python3.5/contextlib.py" in inner
  30.                 return func(*args, **kwds)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/contrib/admin/options.py" in changeform_view
  1449.                 self.save_model(request, new_object, form, not add)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/contrib/admin/options.py" in save_model
  1007.         obj.save()

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/chroniker/models.py" in save
  854.             for o in qs:

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/models/query.py" in __iter__
  256.         self._fetch_all()

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/models/query.py" in _fetch_all
  1087.             self._result_cache = list(self.iterator())

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/models/query.py" in __iter__
  54.         results = compiler.execute_sql()

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/models/sql/compiler.py" in execute_sql
  835.             cursor.execute(sql, params)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/utils.py" in execute
  79.             return super(CursorDebugWrapper, self).execute(sql, params)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/utils.py" in __exit__
  94.                 six.reraise(dj_exc_type, dj_exc_value, traceback)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/utils/six.py" in reraise
  685.             raise value.with_traceback(tb)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/django/db/backends/mysql/base.py" in execute
  110.             return self.cursor.execute(query, args)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in execute
  250.             self.errorhandler(self, exc, value)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/connections.py" in defaulterrorhandler
  42.         raise errorvalue

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in execute
  247.             res = self._query(query)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in _query
  411.         rowcount = self._do_query(q)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/cursors.py" in _do_query
  374.         db.query(q)

File "/home/devcarl/PycharmProjects/newest_projectaxel/projectaxel/myvenv/lib/python3.5/site-packages/MySQLdb/connections.py" in query
  270.             _mysql.connection.query(self, query)

Exception Type: NotSupportedError at /admin/chroniker/job/add/
Exception Value: (1235, "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")
```

**Comment by chrisspen** [comment]
This is an issue with your version of Django and MySQL. Chroniker doesn't generate any custom SQL. The solution is probably to upgrade either your Django version or MySQL version.

**Comment by chrisspen** [comment]
This actually appears to be an issue with my fix to properly delete old log entries. The SQL generated by Django clearly isn't supported. I'll refactor my ORM usage to output something MySQL supports.

**Comment by chrisspen** [comment]
This should now be fixed in version 1.0.6.
